airflow/api_fastapi/core_api/routes/public/dag_run.py

```diff
@@ -90,21 +93,33 @@ async def patch_dag_run_state(
         raise HTTPException(404, f"Dag with id {dag_id} was not found")
 
     if update_mask:
-        if update_mask != ["state"]:
-            raise HTTPException(400, "Only `state` field can be updated through the REST API")
+        for each in update_mask:
+            if each not in ALLOWED_FIELD_MASK:
+                raise HTTPException(400, f"Invalid field `{each}` in update mask")
     else:
-        update_mask = ["state"]
+        update_mask = ALLOWED_FIELD_MASK
 
     for attr_name in update_mask:
+        attr_value = getattr(patch_body, attr_name)
         if attr_name == "state":
-            state = getattr(patch_body, attr_name)
-            if state == DAGRunPatchStates.SUCCESS:
+            if attr_value is None:
+                raise HTTPException(400, "state cannot be empty when it is included in the update mask")
+            if attr_value == DAGRunPatchStates.SUCCESS:
                 set_dag_run_state_to_success(dag=dag, run_id=dag_run.run_id, commit=True)
-            elif state == DAGRunPatchStates.QUEUED:
+            elif attr_value == DAGRunPatchStates.QUEUED:
                 set_dag_run_state_to_queued(dag=dag, run_id=dag_run.run_id, commit=True)
             else:
                 set_dag_run_state_to_failed(dag=dag, run_id=dag_run.run_id, commit=True)
-
+        elif attr_name == "note":
+            # Once Authentication is implemented in this FastAPI app,
+            # user id will be added when updating dag run note
+            # Refer to https://github.com/apache/airflow/issues/43534
+            if dag_run.dag_run_note is None:
+                dag_run.note = (attr_value, None)
+            else:
+                dag_run.dag_run_note.content = attr_value
```

Comment:
@pierrejeambrun , for some reason, when I set the attr_value here, its not actually updating the content on DB. On line 123, dag_run object I'm fetching using `session.get(DagRun, dag_run.id)` still have older dag run note. I verified by debugging that my test case is indeed reaching this line to set the content. Would appreciate some direction here. Am I doing it wrong?

Reply:
You can take a look at the legacy endpoint to see how we set the note. I don’t have access to the code but I remember it using a tuple on the association proxy of the DagRun maybe that can help.

I am out of office atm, let me know if you find a solution.

Reply:
I did refer to legacy implementation and we are doing the exact same thing there. I'll look into this and post here If I find anything

Reply:
Check the comment below, this is due to sessions and not operating on the same object for the two updates (states + note, references are actually different and not operating on the same object).

Normally if we have 1 unique session we could just `refresh` and it should work.